### PR TITLE
Added variable for smtp uri

### PR DESCRIPTION
**Pull request by virajmparekh**
Users will put in an SMTP uri to use for platform invite and alerts

### CODE REVIEW

**Review by sjmiller609 — CHANGES_REQUESTED** [review]
You need to also take this variable and put it into the helm config by modifying locals.tf

File: variables.tf [hunk 1]
```diff
@@ -56,4 +56,6 @@ variable "enable_gvisor" {
   type        = string
 }
 
-
+variable "smtp_uri" {
+  type        = string
```
Comment (sjmiller609) [inline variables.tf]: default = ""

File: locals.tf [hunk 2]
```diff
@@ -44,6 +44,9 @@ nginx:
 astronomer:
   houston:
     config:
+      email:
```
Comment (sjmiller609) [inline locals.tf]: can we make this block conditional similar to how this one is done?
https://github.com/hashicorp/terraform-guides/blob/master/infrastructure-as-code/terraform-0.12-examples/new-template-syntax/rigged_vote.txt
i.e. if var.smtp_uri is "", then omit this block

**Review by sjmiller609 — CHANGES_REQUESTED** [review]
Thanks for adding this, just a couple more things. Sorry for the delay on this review.

**Comment by vparekh94** [comment]
Think I got the formatting piece right, but let me know. Think i got most of the changes from master on the locals.tf as well in that last commit

File: locals.tf [hunk 3]
```diff
@@ -40,14 +36,14 @@ nginx:
...
         value: gvisor
-
 astronomer:
   houston:
     config:
-      commander:
-        enabled: false
-      deployments:
-        logHelmValues: true
+    %{if var.smtp_uri == "true"}
```
Comment (sjmiller609) [inline locals.tf]: if var.smtp_uri != ""

File: locals.tf [hunk 4]
```diff
@@ -40,14 +36,14 @@ nginx:
...
+        email:
+          enabled: true
+          smtpUrl: ${var.smtp_uri == "" ? "" : var.smtp_uri}
```
Comment (sjmiller609) [inline locals.tf]: var.smtp_uri (no conditional here, redundant with above)

File: locals.tf [hunk 5]
```diff
@@ -40,14 +36,14 @@ nginx:
...
+          %{endif}
```
Comment (sjmiller609) [inline locals.tf]: this needs ought to be the same indentation level as the opening "if" line - it looks more indented to me. check for tabs vs spaces, I think we only use spaces.

File: locals.tf (hunk 3 above)
Comment (sjmiller609) [reply locals.tf]: smtp will be a URL, not "true"

**Review by sjmiller609 — CHANGES_REQUESTED** [review]
A few more things

File: locals.tf (hunk 4 above)
Comment (vparekh94) [reply locals.tf]: Good catch!

File: locals.tf [hunk 6]
```diff
@@ -40,14 +36,14 @@ nginx:
...
   houston:
     config:
-      commander:
-        enabled: false
-      deployments:
-        logHelmValues: true
+    %{if var.smtp_uri != ""}
+        email:
+          enabled: true
+          smtpUrl: var.smtp_uri
```
Comment (sjmiller609) [inline locals.tf]: since we are in a string block, you've got to wrap this in ${}
```
          smtpUrl: ${var.smtp_uri}
```
see 'baseDomain' above

**Review by sjmiller609 — CHANGES_REQUESTED** [review]
Looks good, thank you @vparekh94 one more change...

**Review by sjmiller609 — APPROVED** [review]
looks good

**Review by sjmiller609 — CHANGES_REQUESTED** [review]
please resolve merge conflicts

File: locals.tf [hunk 7]
```diff
@@ -40,10 +36,17 @@ nginx:
...
   houston:
     config:
+<<<<<<< HEAD
+    %{if var.smtp_uri != ""}
+        email:
+          enabled: true
+          smtpUrl: ${var.smtp_uri}
+    %{endif}
+=======
+>>>>>>> master
```
Comment (sjmiller609) [inline locals.tf]: nope

**Review by sjmiller609 — CHANGES_REQUESTED** [review]

File: locals.tf (hunk 7 above)
Comment (vparekh94) [reply locals.tf]: I used my atom's GUI git thing to get rid of that - not sure how it got back in - just resolved.

**Review by sjmiller609 — APPROVED** [review]
🎉

**Comment by sjmiller609** [comment]
https://github.com/astronomer/terraform-kubernetes-astronomer/releases/tag/1.0.4

**Pull request merged**